src/WebJobs.Script.WebHost/Management/InstanceManager.cs

```diff
@@ -54,8 +54,17 @@ public bool StartAssignment(HostAssignmentContext context)
                     _assignmentContext = context;
                 }
 
-                // start the specialization process in the background
                 _logger.LogInformation("Starting Assignment");
+
+                // set a flag which will cause any incoming http requests to buffer
+                // until specialization is complete
+                // the host is guaranteed not to receive any requests until AFTER assign
+                // has been initiated, so setting this flag here is sufficient to ensure
+                // that any subsequent incoming requests while the assign is in progress
+                // will be delayed until complete
+                _webHostEnvironment.DelayRequests();
```

Comment:
Some of the failures we're seeing are due to race conditions in the Assign call. Once assign returns, requests for the specialized app can start coming in. The issue was we were returning 202 Accepted and returning, but the actual Assignment was spun up in the background. That leaves a small window of time where a request could come in before the Delay call, resulting in 404s. That would happen very rarely in production, but these tests would reproduce this more often since the calls are back to back.